src/main/java/org/springframework/data/redis/cache/DefaultRedisCacheWriter.java

```diff
@@ -545,28 +539,52 @@ private Mono<Void> doUnlock(String name, ReactiveRedisConnection connection) {
 
 		private Mono<Void> waitForLock(ReactiveRedisConnection connection, String cacheName) {
 
-			AtomicLong lockWaitTimeNs = new AtomicLong();
-			byte[] cacheLockKey = createCacheLockKey(cacheName);
+			AtomicLong lockWaitNanoTime = new AtomicLong();
+
+			Consumer<org.reactivestreams.Subscription> setNanoTimeOnLockWait = subscription ->
+					lockWaitNanoTime.set(System.nanoTime());
 
-			Flux<Long> wait = Flux.interval(Duration.ZERO, sleepTime);
-			Mono<Boolean> exists = connection.keyCommands().exists(ByteBuffer.wrap(cacheLockKey)).filter(it -> !it);
+			Consumer<SignalType> recordStatistics = signalType ->
+					statistics.incLockTime(cacheName, System.nanoTime() - lockWaitNanoTime.get());
 
-			return wait.doOnSubscribe(subscription -> lockWaitTimeNs.set(System.nanoTime())) //
-					.flatMap(it -> exists) //
-					.doFinally(signalType -> statistics.incLockTime(cacheName, System.nanoTime() - lockWaitTimeNs.get())) //
+			Function<Long, Mono<Boolean>> doWhileCacheLockExists = lockWaitTime -> connection.keyCommands()
+					.exists(toCacheLockKey(cacheName)).filter(cacheLockKeyExists -> !cacheLockKeyExists);
+
+			return waitInterval(sleepTime) //
+					.doOnSubscribe(setNanoTimeOnLockWait) //
+					.flatMap(doWhileCacheLockExists) //
+					.doFinally(recordStatistics) //
 					.next() //
 					.then();
 		}
 
+		private Flux<Long> waitInterval(Duration period) {
+			return Flux.interval(Duration.ZERO, period);
+		}
+
+		private ByteBuffer toCacheLockKey(String cacheName) {
+			return ByteBuffer.wrap(createCacheLockKey(cacheName));
+		}
+
+		private ReactiveRedisConnectionFactory getReactiveConnectionFactory() {
+			return (ReactiveRedisConnectionFactory) DefaultRedisCacheWriter.this.connectionFactory;
+		}
+
+		private Mono<ReactiveRedisConnection> getReactiveConnection() {
+			return Mono.fromSupplier(getReactiveConnectionFactory()::getReactiveConnection);
+		}
+
```

Comment:
I disagree. They help reduce the noise in the method where this code is clumped up, thereby causing the code formatter to break lines at ridiculous points. Additionally, using a named method is much clearer than some arbitrary variable, like `cf`.

Reply:
In these methods... 

With `waitForInterval(period)`, it is more descriptive than `Flux.interval(Duration.ZERO, period)`. What is `Duration.ZERO` anyway? Oh right, delay. Why do I even need to know it is `Flux` for that matter? It is an implementation detail. You don't know what `Object.wait(..)` does when you call it.

`getReactiveConnectionFactory()` avoid the cast could serve as an extension point if `protected`. In too many places, we are referring to state variables directly rather than the appropriate getter (where applicable), preventing extensibility.

`getReactiveConnection()` should go without saying, `Mono` into a `connectionFactory.getConnection()`. Having this logic spread all over the class is redundant and busy creating noise in that need wrapped connections.

What does it matter whether it is a variable or a method. Methods help remove clutter from other methods and can be reused.

Reply:
What was there [before](https://github.com/spring-projects/spring-data-redis/pull/2744/files#diff-6c6aae73dbe5cee3bc00315ecfb74e08fb1606bc25c1abe443b2c14afea8e6c0L564-R587) was certainly less readable.